src/lib/plugin.js

```diff
@@ -35,7 +35,10 @@ function * requestRetry (opts, errorMessage, credentials) {
         {json: true},
         requestCredentials(credentials),
         opts))
-      if (res.statusCode >= 400 && res.statusCode < 500) {
+      if (res.statusCode === 404) {
+        debug('request status 404.')
+        break // don't retry if it's a 404
+      } else if (res.statusCode >= 400 && res.statusCode < 500) {
```

Comment:
As a general rule, you should retry on 5xx (server messed up, maybe it'll be fixed later) and not retry on 4xx (you messed up, unless you changed something, probably no point in just trying again).